packages/core/halo/credentials/src/presentations/verifier.ts

```diff
@@ -0,0 +1,73 @@
+//
+// Copyright 2022 DXOS.org
+//
+
+import { verifySignature } from '@dxos/crypto';
+import { Presentation, Proof } from '@dxos/protocols/proto/dxos/halo/credentials';
+
+import { SIGNATURE_TYPE_ED25519, VerificationResult, verifyCredential } from '../credentials';
+import { getPresentationProofPayload } from './signing';
+
+export const verifyPresentation = async (presentation: Presentation): Promise<VerificationResult> => {
+  const errors: string[] = [];
+  const credentialsVerifications = await Promise.all(
+    presentation.credentials?.map((credential) => verifyCredential(credential)) ?? []
+  );
+  for (const verification of credentialsVerifications) {
+    if (verification.kind === 'fail') {
+      errors.push(...verification.errors);
+    }
+  }
+
+  const proofVerification = await Promise.all(
+    presentation.proofs?.map((proof) => {
+      if (!proof.chain) {
+        return {
+          kind: 'fail',
+          errors: ['Delegated credential is missing credential chain.']
+        };
+      }
+      // TODO(mykola): Add chain verification?
```

Comment:
you could use the same code as in credential verification

Reply:
I will add it in a separate PR